provisioning/device/samples/ProvisioningDeviceClientX509/readme.md

```diff
@@ -0,0 +1,53 @@
+# Provisioning Device Client Sample - TPM Attestation
+
+## Overview
+
+This is a quick tutorial with the steps to register a device in the Microsoft Azure IoT Hub Device Provisioning Service using X509 certificates locally generated.
+
+## How to run the sample
+
+Ensure that all prerequisite steps presented in [samples](..\) have been performed.
+
+The sample code is set up to use X509 certificates stored within a password-protected PKCS12 formatted file (certificate.pfx). To generate a self-signed certificate run the following command:
+
+`powershell .\GenerateTestCertificate.ps1`
+
+The script will prompt for a PFX password. The same password must be used when running the sample.
+
+In your Device Provisioning Service go to "Manage enrollments" and select "Individual Enrollments".
+Select "Add" then fill in the following:
+Mechanism: X.509
+Certificate: Select the public key 'certificate.cer' file.
+DeviceID: iothubx509device1 (must match the CN part of the certificate Subject)
```

Comment:
This is incorrect. The device ID does NOT have to match the CN of the cert - that's the registration ID that has to match, and we don't ask for it in the portal because we pull it from the cert automagically. The device ID is an optional field.

Reply:
@nberdy I never tested that scenario - I assume that's only true for Individual Enrollments (where you can specify it).
In Enrollment Groups I've tested that above is always true and there is no way to change the device ID.

/cc @simonporter

Reply:
Will change the comment here since this is only about Individual Enrollments but, if you can confirm, I'll also add the group explanation.

Reply:
I confirm - I have an individual enrollment with a device ID set to something that is not the CN.

In enrollment groups, you cannot specify the device ID because it's a group of devices and we don't yet have the ability to specify a naming rule for the group.